### DISCUSSION TITLE
How to check if pdf is PDF/A or not? And is there a way to convert PDF->PDF/A and vice versa?

### DISCUSSION THREAD

**Discussion by PasaOpasen** [pymupdf/PyMuPDF#2169] [Q&A] [answered]
As I see, mostly PDF/A is PDF + fonts. Is there a way to detect PDF/A and to convert from/to PDF/A?

**Comment by JorjMcKie** [comment, marked as answer]
> As I see, mostly PDF/A is PDF + fonts. Is there a way to detect PDF/A and to convert from/to PDF/A?

Well, that's a simplified characterization.
Anyway, PyMuPDF does not support PDF/A on output and saves in standard format. Cannot **_convert to_** PDF/A - and there is no intention to do so in the foreseeable future.

To determine if an **_input_** file is in some of those formats (yes: plural!), check the catalog. It will contain an "output intent dictionary", see page 641 / "Table 365" ("Document management - Portable document format - Part 1: PDF 1.7") for a description. To access it, get the catalog `xref = doc.pdf_catalog()` and then inspect `doc.xref_get_key(xref, "OutputIntents")`. This should be an array containing a list of one or more xrefs pointing to output intent dictionaries.

**Reply by PasaOpasen** [reply]
@JorjMcKie `see page 641 / "Table 365" ("Document management - Portable document format - Part 1: PDF 1.7")` what document do u mean?

**Reply by JorjMcKie** [reply]
theone referenced in the pymupdf documentation https://pymupdf.readthedocs.io/en/latest/app3.html#adobe-pdf-references

**Reply by PasaOpasen** [reply]
@JorjMcKie Thank u! So, again:
- there is no  way to PDF->PDF/A conversion (with fitz)
- it is possible to check if document is like PDF/A by `OutputIntents`
- what about PDF/A->PDF conversion?

**Reply by JorjMcKie** [reply]
as I wrote: on saving any PDF, non-PDF/* will be created.

**Comment by indigoviolet** [comment]
@JorjMcKie 

>Anyway, PyMuPDF does not support PDF/A on output and saves in standard format. Cannot convert to PDF/A - and there is no intention to do so in the foreseeable future.

Do you have any suggestions for how to convert PDFs to PDF/A programmatically? In my experience, Ghostscript conversion doesn't work well (I see missing text). The alternative I've found is Apryse which does work well for my small sample of texts, but we would ideally like to standardize on one PDF library. We already license PyMuPDF, so this would be a great fit for this problem as well.

**Reply by JorjMcKie** [reply]
This would have to be enabled by our base library MuPDF.
May I recommend you reach out to the developers in [MuPDF's](https://discord.gg/DajS59Z2) public Discord channel?
